CoffeeShop/src/main/java/client/CustomerController.java

```
@@ -0,0 +1,166 @@
+        orders = OrderList.getInstance();
+        menu = ItemList.getInstance();
+
+        // Initialize the current order
+        try {
+            currentOrder = new Order();
+        } catch (InvalidOrderException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Set up action listeners for buttons
+        view.getSubmitOrderButton().addActionListener(this);
+        view.getCancelOrderButton().addActionListener(this);
+        view.getAddItemButton().addActionListener(this);
+        view.getRemoveLastItemButton().addActionListener(this);
+        view.getRemoveItemButton().addActionListener(this);
+        view.getExitButton().addActionListener(this);
+
+        // Display the menu in the View
+        view.displayMenu(menu.getMenuDetails());
+    }
+
+    /**
+     * Handles button clicks
+     *
+     * @param e The ActionEvent
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getSource() == view.getSubmitOrderButton()) {
+            submitOrder();
+        } else if (e.getSource() == view.getCancelOrderButton()) {
+            cancelOrder();
+        } else if (e.getSource() == view.getAddItemButton()) {
+            addItem();
+        } else if (e.getSource() == view.getRemoveLastItemButton()) {
+            removeLastItem();
+        } else if (e.getSource() == view.getRemoveItemButton()) {
+            removeItem();
+        } else if (e.getSource() == view.getExitButton()) {
+            JOptionPane.showMessageDialog(view, "Good Bye!");
+            exit();
+        }
+    }
+
+    /**
+     * Submits the current order
+     */
+    private void submitOrder() {
+        try {
+            if (orders.add(currentOrder)) {
+                Demo.demoWriteOrders();
+                JOptionPane.showMessageDialog(view, "Order has been submitted");
+            } else {
+                JOptionPane.showMessageDialog(view, "Order could not be placed - Please Try Again Later");
+            }
```

```suggestion
            if (!orders.add(currentOrder)) {
                JOptionPane.showMessageDialog(view, "Order could not be placed - Please Try Again Later");
                return;
            }
            
            Demo.demoWriteOrders();
            JOptionPane.showMessageDialog(view, "Order has been submitted");
```

---

we want to fail fast